dissect/target/tools/query.py

```diff
@@ -110,41 +118,78 @@ def main():
-
-        if not obj:
-            parser.error(f"Can't find plugin with function `{first_func}`")
-
+        found_functions = find_plugin_functions(Target(), args.function, False)
+        if not len(found_functions):
+            parser.error("Function(s) not found, see -l for available plugins")
+            parser.exit()
+        func = found_functions[0]
+        if issubclass(func.class_object, OSPlugin):
+            func.class_object = OSPlugin
+        obj = getattr(func.class_object, func.method_name)
         if isinstance(obj, type) and issubclass(obj, Plugin):
             parser = generate_argparse_for_plugin_class(obj, usage_tmpl=USAGE_FORMAT_TMPL)
+        elif isinstance(obj, Callable) or isinstance(obj, property):
+            parser = generate_argparse_for_unbound_method(getattr(obj, "fget", obj), usage_tmpl=USAGE_FORMAT_TMPL)
         else:
-            parser = generate_argparse_for_unbound_method(obj, usage_tmpl=USAGE_FORMAT_TMPL)
-
+            parser.error(f"Can't find plugin with function `{func.method_name}`")
         parser.print_help()
         parser.exit()
 
-    if not args.targets:
-        parser.error("too few arguments")
+    # Show the list of available plugins for the given optional target and optional
+    # search pattern, only display plugins that can be applied to ANY targets
+    if args.list:
+        collected_plugins = {}
+
+        if args.targets:
+            for target in args.targets:
+                plugin_target = Target.open(target)
+                funcs = find_plugin_functions(plugin_target, args.function, True)
+                for func in funcs:
+                    collected_plugins[func.name] = func.plugin_desc
+        else:
+            funcs = find_plugin_functions(Target(), args.function, False)
+            for func in funcs:
+                collected_plugins[func.name] = func.plugin_desc
+
+        # Display in a user friendly manner
+        target = Target()
+        fparser = generate_argparse_for_bound_method(target.plugins, usage_tmpl=USAGE_FORMAT_TMPL)
+        fargs, rest = fparser.parse_known_args(rest)
+
+        if len(collected_plugins):
+            target.plugins(list(collected_plugins.values()))
+
+        # No real targets specified, user might be interested to see available loaders
+        if not args.targets:
+            fparser = generate_argparse_for_bound_method(target.loaders, usage_tmpl=USAGE_FORMAT_TMPL)
+            fargs, rest = fparser.parse_known_args(rest)
+            target.loaders(**vars(fargs))
+        parser.exit()
+
+    # Verify uniformity of output types, otherwise default to records
+    output_types = set()
+    funcs = find_plugin_functions(Target(), args.function, False)
+    for func in funcs:
+        output_types.add(func.output_type)
+
+    default_output_type = None
+    if len(output_types) > 1:
+        log.warning("Mixed output types detected: %s, only outputting records.", "/".join(output_types))
```

Comment:
```suggestion
        log.warning("Mixed output types detected: %s. Only outputting records.", ",".join(output_types))
```
"," makes more sense in my opinion.